src/tangle/cone_stream.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::{stream::BoxStream, Stream};
+use pin_project::pin_project;
+
+use super::{ledger_updates::LedgerUpdateStore, InputSource};
+use crate::types::{
+    ledger::BlockMetadata,
+    stardust::block::{payload::TransactionEssence, Block, BlockId, Input, Output, Payload},
+};
+
+/// A [`Block`] enriched with [`BlockMetadata`] and potentially [`Output`]s that correspond to inputs.
+pub struct BlockWithMetadataInputs {
```

Comment:
not very constructive feedback I know but those more and more frankenstein'ed struct names indicate that we're not modelling this right IMO.

Reply:
The problem is that we're mixing two layers here: the `Block` layer and the `Transaction` layer (that builds on top of the `Block` layer), and I am not sure we should do that.

Reply:
Wouldn't it be better to have for example a `TransactionStream` type, that only streams all `Transactions` with their inputs and outputs and relevant metadata as they occured within a milestone?